### ISSUE TITLE
Calling into a component that includes the println macro hangs

### ISSUE THREAD

**Issue by DamianReeves** [bytecodealliance/jco#449]
Calling into the component built [here](https://github.com/DamianReeves/wasm-components-ftw), results in a hang as long as a `println!` is included. 

As seen here:
https://github.com/DamianReeves/wasm-components-ftw/blob/5a85c47fdc62288cb46952484f760231073c5152/crates/morphir-platform/src/lib.rs#L11-L15

If I remove the `println` things work fine.

Not sure if its relevant, but if I transpile with tracing, I notice when I insert a `println!` call, my code seems to hang on the following  wasi call is traces:

```
[module="wasi:cli/stdin@0.2.0", function="get-stdin"] call
```

**Comment by guybedford** [comment]
Thanks for posting. Unfortunately I was unable to build this test case on arm64. Can you just share the direct component binary that is causing the issue?

**Comment by DamianReeves** [comment]
Sure thing: Here you go!

[dist.zip](https://github.com/user-attachments/files/16100410/dist.zip)

**Comment by guybedford** [comment]
It seems you've built the full preview2-shim library into a single file on Node.js. Unfortunately I think this might be breaking some invariants of the syncification that spawns a worker here:

```
var _rawDebug = nodeProcess._rawDebug || console.error.bind(console);
var workerPath = fileURLToPath(new URL("./worker-thread.js", import.meta.url));
var httpIncomingHandlers = new Map;
var instanceId = Math.round(Math.random() * 1000).toString();
var DEBUG_DEFAULT = false;
var DEBUG = env.PREVIEW2_SHIM_DEBUG === "0" ? false : env.PREVIEW2_SHIM_DEBUG === "1" ? true : DEBUG_DEFAULT;
var ioCall = createSyncFn(workerPath, DEBUG, (type, id, payload) => {
```

**Comment by guybedford** [comment]
Yes exactly, or at least not bundle in the worker invocation as that must be a separate file.

**Comment by vados-cosmonic** [comment]
Hey @DamianReeves it's been a while but hopefully this problem was fixed -- please let me know if it wasn't and we'll re-open this.
